odl/discr/grid.py

```diff
@@ -451,54 +532,85 @@ def __contains__(self, other):
                     for vector, coord in zip(self.coord_vectors, other)))
 
     def is_subgrid(self, other, atol=0.0):
-        """Test if this grid is contained in another grid.
+        """Return ``True`` if this grid is a subgrid of ``other``.
 
         Parameters
         ----------
-        other :  `TensorGrid`
-            The other grid which is supposed to contain this grid
+        other :  `RectGrid`
+            The other grid which is supposed to contain this grid.
         atol : float
             Allow deviations up to this number in absolute value
             per coordinate vector entry.
 
+        Returns
+        -------
+        is_subgrid : bool
+            ``True`` if all coordinate vectors of ``self`` are within
+            absolute distance ``atol`` of the other grid, else ``False``.
+
         Examples
         --------
-        >>> g = TensorGrid([0, 1], [-1, 0, 2])
-        >>> g_sub = TensorGrid([0], [-1, 2])
-        >>> g_sub.is_subgrid(g)
+        >>> rg = uniform_grid([-2, -2], [0, 4], (3, 4))
+        >>> rg.coord_vectors
+        (array([-2., -1.,  0.]), array([-2.,  0.,  2.,  4.]))
+        >>> rg_sub = uniform_grid([-1, 2], [0, 4], (2, 2))
+        >>> rg_sub.coord_vectors
+        (array([-1.,  0.]), array([ 2.,  4.]))
+        >>> rg_sub.is_subgrid(rg)
         True
-        >>> g_sub = TensorGrid([0.1], [-1.05, 2.1])
-        >>> g_sub.is_subgrid(g)
+
+        Fuzzy check is also possible. Note that the tolerance still
+        applies to the coordinate vectors.
+
+        >>> rg_sub = uniform_grid([-1.015, 2], [0, 3.99], (2, 2))
+        >>> rg_sub.is_subgrid(rg, atol=0.01)
         False
-        >>> g_sub.is_subgrid(g, atol=0.15)
+        >>> rg_sub.is_subgrid(rg, atol=0.02)
         True
         """
         # Optimization for some common cases
         if other is self:
             return True
-        if not (isinstance(other, TensorGrid) and
+        if not (isinstance(other, RectGrid) and
```

Comment:
well any non rect grid should be false, I was thinking about the following lines but perhaps I missed the logic (hard to read with changelogs)